src/compiler.ts

```diff
@@ -6314,6 +6321,7 @@ export class Compiler extends DiagnosticEmitter {
                 module.local_get(tempThis.index, nativeThisType),
                 nativeFieldType, field.memoryOffset
               ),
+              fieldType,
```

Comment:
Here we are missing `valueType` as well as an argument to `makeFieldAssignment`. Again, we probably don't need it, but perhaps a general convention of having an expression plus a type on all `makeXY` methods makes this clearer and avoids issues?